src/Orleans.Indexing/Indexes/StorageManagedIndexes/DirectStorageManagedIndexImpl.cs

```diff
@@ -42,11 +41,11 @@ public override Task OnActivateAsync()
 
         public Task<bool> DirectApplyIndexUpdateBatch(Immutable<IDictionary<IIndexableGrain, IList<IMemberUpdate>>> iUpdates,
                                                         bool isUnique, IndexMetaData idxMetaData, SiloAddress siloAddress = null)
-            => Task.FromResult(true);
+            => Task.FromResult(true);   // The index is maintained by the underlying _grainStorage, when the grain's WriteStateAsync is called
```

Comment:
After looking at the CosmosDb provider I'm really not sure this is a good idea. Yes, CosmosDb can index, but the current implementation is a stopper when it comes to horizontal scaling over partitions. Other storage providers will have similar problems or must support indexing separately. Having each storage provider extract fields for indexing upon WriteStateAsync seems like a trap. Having explicit calls for indexing to the storage providers will also allow for using a different provider for indexing. Just because CosomsDB can do it in one go should not force all the others to deal with extracting data for indexing.